File: medplum-java/medplum-model/src/main/java/com/fhir/Observation_ReferenceRange.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableObservation_ReferenceRange.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Observation_ReferenceRange {
  Optional<String> text();

  Optional<List<Extension>> extension();

  Optional<Quantity> low();

  Optional<String> id();

  Optional<CodeableConcept> type();

  Optional<Quantity> high();

  Optional<Range> age();

  Optional<List<Extension>> modifierExtension();

  Optional<List<CodeableConcept>> appliesTo();

  static ImmutableObservation_ReferenceRange.Builder builder() {
    return ImmutableObservation_ReferenceRange.builder();
  }
}

File: medplum-typescript/src/model/fhir/Observation_ReferenceRange.ts
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { Quantity } from "./Quantity";
import { Range } from "./Range";

export interface Observation_ReferenceRange {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    low: Quantity | null | undefined;
    high: Quantity | null | undefined;
    type: CodeableConcept | null | undefined;
    appliesTo: CodeableConcept[] | null | undefined;
    age: Range | null | undefined;
    text: string | null | undefined;
}
